src/context/global-state-context.tsx

```diff
@@ -39,6 +40,18 @@ export const useFragment = (id?: string) => {
 
 const GlobalStateContextProvider = ({ children }: any) => {
 	const [fragments, _setFragments] = useState<any[]>(getFragmentsFromLocalStorage());
+
+	// Execute this only once, on initial builder load
+	useEffect(() => {
+		// Check version & migrate if needed!
+		fragments.forEach((frag: any) => {
+			if (frag.version !== CURRENT_MODEL_VERSION) {
+				updateModel(frag);
```

Comment:
This approach is just making use of the existing approach, once you launch/navigate the app, it will get everything from the local storage and it will ALSO save to local storage.

1. Dashboard, calls `updateFragment` in `useEffect` which sets Fragments to an empty state.
2. GlobalStateContextProvider gets all fragment data from local storage (useEffect)
  a) Then `setFragment` is called which saves to the local storage.

Reply:
I noticed there was another useEffect with same dependencies (Empty array so only called once) so I've merged the two useEffects.